packages/maskbook/src/plugins/Wallet/services/tokenPrice.ts

```diff
@@ -0,0 +1,122 @@
+import { CryptoPrice, CurrencyType } from '@masknet/web3-shared'
+import { pollingTask } from '@masknet/shared'
+import { getTokenPrices, getNativeTokenPrice } from '../../../extension/background-script/EthereumService'
+import { startEffects } from '../../../utils'
+import { UPDATE_CHAIN_STATE_DELAY } from '../constants'
+import { currentTokenPricesSettings } from '../settings'
+import { uniq } from 'lodash-es'
+
+function updateCurrentPrices(data: CryptoPrice) {
+    const currentPrices: CryptoPrice = { ...currentTokenPricesSettings.value }
+    Object.keys(data).forEach((category) => {
+        Object.keys(data[category]).forEach((currency) => {
+            currentPrices[category] = {
+                ...currentPrices[category],
+                [currency]: data[category][currency],
+            }
+        })
+    })
+    currentTokenPricesSettings.value = currentPrices
+}
+
+let tokenTrackingCount = 0
+let nativeTokenCount = 0
+
+export async function kickToUpdateTokenPrices() {
+    tokenTrackingCount += 1
+    nativeTokenCount += 1
+}
```

Comment:
This function does not have any asynchronous tasks in it

Reply:
Right, just removed `async`.